tests/objcthemis/objthemis/SecureMessageTests.m

```diff
@@ -37,55 +37,105 @@ - (void)testGeneratingKeys {
 }
 
 - (void)testSecureMessageRSA1024 {
-    [self wrapUnwrapWithPrivateKey:privateKeyRSA1024 publicKey:publicKeyRSA1024];
-    [self signVerifyWithPrivateKey:privateKeyRSA1024 publicKey:publicKeyRSA1024];
+    [self wrapUnwrapWithPrivateStringKey:privateKeyRSA1024 publicStringKey:publicKeyRSA1024];
+    [self signVerifyWithPrivateStringKey:privateKeyRSA1024 publicStringKey:publicKeyRSA1024];
 }
 
 - (void)testSecureMessageRSA2048 {
-    [self wrapUnwrapWithPrivateKey:privateKeyRSA2048 publicKey:publicKeyRSA2048];
-    [self signVerifyWithPrivateKey:privateKeyRSA2048 publicKey:publicKeyRSA2048];
+    [self wrapUnwrapWithPrivateStringKey:privateKeyRSA2048 publicStringKey:publicKeyRSA2048];
+    [self signVerifyWithPrivateStringKey:privateKeyRSA2048 publicStringKey:publicKeyRSA2048];
 }
 
 - (void)testSecureMessageRSA4096 {
-    [self wrapUnwrapWithPrivateKey:privateKeyRSA4096 publicKey:publicKeyRSA4096];
-    [self signVerifyWithPrivateKey:privateKeyRSA4096 publicKey:publicKeyRSA4096];
+    [self wrapUnwrapWithPrivateStringKey:privateKeyRSA4096 publicStringKey:publicKeyRSA4096];
+    [self signVerifyWithPrivateStringKey:privateKeyRSA4096 publicStringKey:publicKeyRSA4096];
 }
 
+- (void)testSecureMessageKeygenRSA {
+  TSKeyGen *keygenRSA = [[TSKeyGen alloc] initWithAlgorithm:TSKeyGenAsymmetricAlgorithmRSA];
+  [self wrapUnwrapWithPrivateKey:keygenRSA.privateKey publicKey:keygenRSA.publicKey];
+  [self signVerifyWithPrivateKey:keygenRSA.privateKey publicKey:keygenRSA.publicKey];
+}
+
+- (void)testSecureMessageKeygenEC {
+  TSKeyGen *keygenEC = [[TSKeyGen alloc] initWithAlgorithm:TSKeyGenAsymmetricAlgorithmEC];
+  [self wrapUnwrapWithPrivateKey:keygenEC.privateKey publicKey:keygenEC.publicKey];
+  [self signVerifyWithPrivateKey:keygenEC.privateKey publicKey:keygenEC.publicKey];
+}
 
-- (void)wrapUnwrapWithPrivateKey:(NSString *)privateKey publicKey:(NSString *)publicKey {
-    NSData *publicKeyData = [[NSData alloc] initWithBase64EncodedString:publicKey
-                                                                options:NSDataBase64DecodingIgnoreUnknownCharacters];
-    NSData *privateKeyData = [[NSData alloc] initWithBase64EncodedString:privateKey
-                                                                 options:NSDataBase64DecodingIgnoreUnknownCharacters];
 
+- (void)wrapUnwrapWithPrivateStringKey:(NSString *)privateKey publicStringKey:(NSString *)publicKey {
+  NSData *publicKeyData = [[NSData alloc] initWithBase64EncodedString:publicKey
+                                                              options:NSDataBase64DecodingIgnoreUnknownCharacters];
+  NSData *privateKeyData = [[NSData alloc] initWithBase64EncodedString:privateKey
+                                                               options:NSDataBase64DecodingIgnoreUnknownCharacters];
+  
+  return [self wrapUnwrapWithPrivateKey:privateKeyData publicKey:publicKeyData];
+}
+
+- (void)wrapUnwrapWithPrivateKey:(NSData *)privateKeyData publicKey:(NSData *)publicKeyData {
```

Comment:
I guess we can use `encryptDecryptWithPrivateKey` instead of `wrapUnwrap` now, if the other one is `signVerify`.